aten/src/ATen/native/Repeat.cpp

```diff
@@ -37,7 +43,17 @@ Tensor repeat_interleave(const Tensor &self, const Tensor &repeats, c10::optiona
         AT_ERROR("repeats must be 0-dim or 1-dim tensor");
     }
 
-    return input.index_select(dim.value(), at::repeat_interleave(repeats_));
+    Tensor indices = at::repeat_interleave(repeats_);
+
+    Tensor indices_sizes = at::ones({self.dim()}, TensorOptions(at::kLong));
```

Comment:
Why are you storing shape as tensor? Is this approach faster?

Reply:
How about 
```
std::vector<int64_t> view_sizes(self.dim(),1);
view_sizes[dim.value()]=indices.size(0);
std::vector<int64_t> expand_sizes = self.size();
expand_sizes[dim.value()]=indices.size(0);
indices.view(view_sizes).expand(expand_sizes);
```
This avoids sizes as tensor things that are in general less efficient than std::vector or better yet SmallVector.

Reply:
Good point. I replaced the indices with simple vectors and it seemed to shave a bit of overhead, though it's hard to measure. Overall I think it's cleaner, though.

Reply:
I saw this after responding. (the git GUI didn't show Natalia's comment.) I think there are some improvements over what I came up with. (using the view and also using SmallVector)

Reply:
Ok, vector take 2 is done. SmallVector may be a bit faster, but I can't say for certain since it's within system variation.